pkg/patterns/declarative/reconciler.go

```diff
@@ -302,18 +302,20 @@ func (r *Reconciler) reconcileExists(ctx context.Context, name types.NamespacedN
 		}
 	}
 
-	applyOpt := applier.ApplierOptions{
-		RESTConfig: r.config,
-		RESTMapper: r.restMapper,
-		Namespace:  ns,
-		Manifest:   manifestStr,
-		Validate:   r.options.validate,
-		ExtraArgs:  extraArgs,
-		Force:      true,
+	applierOpt := applier.ApplierOptions{
+		RESTConfig: 		r.config,
+		RESTMapper: 		r.restMapper,
+		Namespace:  		ns,
+		Manifest:   		manifestStr,
+		Validate:   		r.options.validate,
+		ExtraArgs:  		extraArgs,
+		Force:      		true,
+		// TODO Make this configurable
+		CascadingStrategy: 	"Foreground",		
```

Comment:
I'd really like to have this configurable. ATM we don't even know what we wanna have here so we might not be happy with "Foreground" going forward.
I first thought I could create a new interface like `PruneWhiteLister`, but this is maybe something for another PR? And then I wonder if those Interfaces are the way to go?
Would it be possible to have a more general approach to allow to configuration of more of those fields and would that make sense?

Reply:
Putting it into the ApplierOptions is a good start :-)  We should think about mapping this field in the kubectl applier though, as in theory people are still using that (and once the applyset applier is more ready we would do that, but right now I don't think that supports prune anyway)

We could have something like `r.options.validate`, but we could also think about creating a set of the configurable apply options.  I have a PR out for "hooks" that I think would allow this to be changed through code, but as we control the implementation here we can choose what user-experience (developer-experience) we want to create.

I think what we would ideally do is wait till the objects are actually deleted - I don't know if _any_ of the cascadingstrategies actually do that (?), but kubectl can wait so it should be possible...

Reply:
About the cascading strategy, reading [the docs](https://kubernetes.io/docs/concepts/architecture/garbage-collection/#cascading-deletion).
`Background` is the default usually (however there was no default set when I ran it, it resulted in an error complaining that the propagationPolicy which ultimately is the cascading strategy can't be `/`). 
I do believe that we in our controller want to wait and agree with you that this makes probably the most sense to be the default. And I believe that it is what the `Foreground` strategy does (from docs: `The object remains visible through the Kubernetes API until the deletion process is complete.`)
I haven't been able to really see the difference when trying it out yet.

Reply:
I understand that we probably also need to make some changes in the other applies, `exec.go`!? I don't really know if you want like to see this changes in this PR already. Will have a look at least :)

Reply:
Looking at that applier I see multiple reasons why pruning won't work. Some of which I fixed for the direct applier, some more. On the one hand I feel that I know a decent amount about it now to probably fix all of it quite swiftly, but does it make sense?  I could take half an hour or so to make a draft PR building on top of this to do everything I know to at least have a quick start?

Reply:
I have no idea how the `manifest with prune` test in `exec_test.go` can even work :D
For sure the `expectedArgs` will not result in the expected behavior (pruning). As you can't use `-f` and `--prune`.

That was one of my confusions in the direct applier. If you set force in the direct applier it is not gonna result in the `-f` behaviour but it results in setting `ForceConflicts` aka `--force-conflict`. Thinking about this I believe this is quite wrong